### ISSUE TITLE
Request for Grade

### ISSUE THREAD

**Issue by mmcgreg** [mmcgreg/web-design-2-independent-project#10]
After reviewing the rubric and looking over how the points are allocated I feel comfortable saying that this website is somewhere between a B+ and A- grade. The reason for the range is that I am still learning to understand and implement certain semantics in html as well as the best way to use CSS to attain my websites end goals. Performance testing is also something that still mostly perplexes me. I understand the reasoning behind it but I cannot with much certainty explain what a lot of it actually means.

I have submitted with Markbot passing all checks and getting 33/33 for accessibility tests that it runs.
I have also done my own user testing as well as reviewed my peers issues from last class and made the necessary changes.

**TLDR;** 
Overall, this was a challenge but one I enjoyed. I fully grasp the necessary base knowledge to code any basic website with HTML and CSS but the nuance of CSS sometimes still takes time for me. I'm proud that I was able to more or less successfully implement responsiveness into my website and am happy with the overall layout. I tackled the challenge of cards which is something that when first covered this semester really escaped me. I'm happy to see it having worked for me this time around, and feel more confident implementing it in the future. Instant recall of information is still not always forthcoming, I consistently referred to the cheat sheets and videos throughout coding, but that is why we have them, so I cross that off as using the material provided and general resourcefulness!

**Knowledge:** 

- I struggled with card layouts and made a point to tackle that in this assignment. I believe I have done a very good job of using cards in my site and I'm pleased with their responsiveness. I realized quickly (though not as quickly as I would like) that my card images were very slightly different sizes even though I had brought them all down to the same size using photoshop and then exporting and smushing them. The few millimetre difference was causing the cards to flow and cascade at different rates as the screen size changed. I had to start from scratch with the original .jpg files and go through the compression and resizing process again for each, but it worked!

**Responsiveness:**

- A little awkwardness in the responsiveness. I struggled with the body text and paragraph tags throughout the first 2 pages of the site. Finding the sweet spot was difficult but I believe I was able to more or less find a happy medium, though it may  be a little jittery at times. 
- The cards on page three are by far the best use of responsiveness, as the CSS flex used, made it so that no matter the screen size they were well laid out and responsive.

**Semantics:**

- My HTML is semantically well done for the most part. When viewed without CSS the HTML is successfully used to differentiate between page sections. The class tags used are relevant to the information they are referring to and made it easy for me to target them in CSS.
- In general I have struggled with redundant code throughout the semester. I know for a fact that there is probably a bit of redundant code in my CSS but fortunately nothing that seems to be truly effecting the site itself or the site at any particular media query size. 

**Design:**

- My design is fairly straightforward and clear. I made an effort to implement good use of layout and with that came legibility. Though it may not be the prettiest website it is functional and well thought out. The information is clear and easily accessed. The colours are harmonious and particular attention was given to readability with contrast. The gray background was chosen with eye strain in mind.

**Accessibility:**

- Running through the website with VoiceOver for Mac shows that it is accessible to the blind and those with limited vision. 
- Skip links are functional and appropriate aria labels and alt states were added to the needed tags and images. They are recognized by the screen reader when used in testing.
- <id> tags were used to associate the relevant skip link to section on the pages, and they are functional.
- The website passed 33/33 of markbot's accessibility tests on submission.
- Colour contrast was also taken into consideration for darker or lighter screens. The use of black text allows for a wide range of readability at different contrast settings.

**Performance:**

- The site was tested in Firefox, Chrome, Opera, Safari, and on mobile Chrome and Safari. No issues found on any browsers.
- I am still running into a failing grade for cache static on my images specifically. I do not recall covering that in class at all and when looking up the fixes it was outside of my knowledge on how to exactly implement them.
- I believe for what we have covered this semester as far as performance testing goes I have hit the mark.

**Git & Commits:**

- I believe I fully understand how to properly commit with github. I do need to work on the message content a little though they are 9/10 times semantically correct and no errors are given in markbot for the verbiage used.
- More commits could be used on a site this size. I unfortunately tend to fall into a coding black hole and come back up for air realizing I have committed in a while. I could implement a timer maybe...
